source/annotations/helpers.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+func getAliasFromAnnotations(annotations map[string]string) bool {
+	aliasAnnotation, exists := annotations[AliasAnnotationKey]
+	return exists && aliasAnnotation == "true"
+}
+
+// TTLFromAnnotations TODO: copied from source.go. Refactor to avoid duplication.
+// TTLFromAnnotations extracts the TTL from the annotations of the given resource.
+func TTLFromAnnotations(annotations map[string]string, resource string) endpoint.TTL {
+	ttlNotConfigured := endpoint.TTL(0)
+	ttlAnnotation, exists := annotations[TtlAnnotationKey]
+	if !exists {
+		return ttlNotConfigured
+	}
+	ttlValue, err := parseTTL(ttlAnnotation)
+	if err != nil {
+		log.Warnf("%s: \"%v\" is not a valid TTL value: %v", resource, ttlAnnotation, err)
+		return ttlNotConfigured
+	}
+	if ttlValue < ttlMinimum || ttlValue > ttlMaximum {
+		log.Warnf("TTL value %q must be between [%d, %d]", ttlValue, ttlMinimum, ttlMaximum)
+		return ttlNotConfigured
+	}
+	return endpoint.TTL(ttlValue)
+}
+
+// parseTTL parses TTL from string, returning duration in seconds.
+// parseTTL supports both integers like "600" and durations based
+// on Go Duration like "10m", hence "600" and "10m" represent the same value.
+//
+// Note: for durations like "1.5s" the fraction is omitted (resulting in 1 second
+// for the example).
+func parseTTL(s string) (ttlSeconds int64, err error) {
```

Comment:
Why define return variables in the signature if you don't use the feature?
Please drop them it's most off the time a code smell.

Reply:
Will do. I was just moved things around, it was like that.

Reply:
current master https://github.com/kubernetes-sigs/external-dns/blob/277e8ca2ec7e10f749083bfd9842d61cc4fd8f32/source/source.go#L115

Reply:
I actually think the error is used on the client. 

If an error, provide default
```
ttlValue, err := parseTTL(ttlAnnotation)
if err != nil {
   return ttlNotConfigured
}
```

Reply:
Pls suggest what to do. Another option is `parseTTLOrDefault`

Reply:
done